google/cloud/storage/internal/service_account_credentials.h

```diff
@@ -145,7 +146,7 @@ class ServiceAccountCredentials : public storage::Credentials {
     }
 
     // TODO(#516) - use retry policies to refresh the credentials.
-    auto response = request_.MakeRequest();
+    auto response = request_.MakeRequest(payload_);
```

Comment:
I'd prefer not to cache the payload after it's initialized for service account credentials (or more generally, cred types that use a JWT for the refresh process).  In the upcoming credential changes I'm working on, I want to provide users with a way to change the JWT attributes (list scopes, "iss" attribute, etc.), which will result in a different payload.  Rather than worry about invalidating `payload_` if an attribute changes, I'd rather we just construct the payload inside of `Refresh()`.  This happens roughly once an hour, so I don't think this will be a big performance hit :P

Thoughts?

Reply:
I think that makes sense, but maybe it should be part of a separate PR?  And note that we were caching it already (as a member variable of `request_`).

Reply:
Oh, and I tried the obvious change for this: move the code from the constructor to `Refresh()` and the unit tests started failing.  I gave up at that point, it is probably easy to get it to work, but really does belong in a separate PR.

Reply:
Oh hey, you're right.  Derp.